### ISSUE TITLE
Question: Support for Egyptian Hieroglyph Format Controls

### ISSUE THREAD

**Issue by Eli-Zaretskii** [harfbuzz/harfbuzz#2727]
Does HarfBuzz support Egyptian Hieroglyph Format Controls, as described in the Unicode Standard section 11.4?

I've tried shaping the examples in the Unicode Standard with the latest `hb-view` v2.7.2, and they produce linear sequence of glyphs, not the arrangement expected from the examples, given the vertical and horizontal formatting controls.

For example, the sequence <13000, 13430, 13250> is supposed to produce the display of one glyph on top of the other (see p.437 of the Unicode Standard), but what I see is a sequence of 3 glyphs for those codepoints, laid out left to right, one after the other, with U+00013430 depicted as an empty rectangle.

This is on MS-Windows 10, with hb-view extracted from the "official" HarfBuzz release on this site (`harfbuzz-win32-2.7.2.zip`), and with the Segoe UI Historical font.  FTR, the command I used was

          hb-view --output-format=svg -o egypt.svg -u "0x13000 0x13430 0x13250" c:\Windows\Fonts\seguihis.ttf

Is this supposed to work?  If the problem is with the font, then what fonts are known to work with this feature?

**Comment by khaledhosny** [comment]
AFAIK, there is nothing (yet) in OpenType for shaping Egyptian Hieroglyphs. Encoding the format controls alone will not provide much without comparable support at font side. Judging by #2558 it seems MS is going with embedding the whole logic in the font itself.

**Comment by alerque** [comment]
It's probably worth mentioning #2721 as well. For that matter [here are all threads mentioning Hieroglyphs](https://github.com/harfbuzz/harfbuzz/issues?q=Hieroglyph).

**Comment by Eli-Zaretskii** [comment]
Thanks.

AFAIU, https://github.com/harfbuzz/harfbuzz/issues/2721 seems to indicate that this *was* working in some version of HarfBuzz?  Because @khaledhosny seems to say that it never actually worked, and the solution is only in the works?  Or what am I missing?

**Comment by khaledhosny** [comment]
The shaping logic is handled entirely by the font, so you need a font that implements this, like the one attached to the issue above. There seem to be some unresolved issues with this font, but this is not directly related to your question.

**Comment by xadxura** [comment]
@Eli-Zaretskii, I've published a project that generates the font logic (in Volt syntax) for a plain Hieroglyphic font here:
https://github.com/microsoft/font-tools/tree/master/EgyptianOpenType
You're welcome to use it. I need to provide more documentation, but haven't had time to do that yet. So please reach out if you have questions.

**Comment by Eli-Zaretskii** [comment]
Thanks.  My level of knowledge in these matters is really quite limited, basically what the Unicode Standard says about this.  What I'd like is to have a font which would work with HarfBuzz and supporte the quadrat arrangements of hieroglyphs per Unicode.  Do I understand correctly that the project you published is capable of producing such fonts?  If so, where are the instructions to produce a font?  Failing that, is there some font I can try?  (I did try the one posted in https://github.com/harfbuzz/harfbuzz/issues/2558, and it didn't quite work, but maybe that was because I did something wrong.  Is that font supposed to handle this?)
